server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -1937,49 +1938,57 @@ public void revokeEntitlements(List<Entitlement> entsToRevoke, Set<String> alrea
 
         log.info("Starting batch delete of pools");
         poolCurator.batchDelete(poolsToDelete, alreadyDeletedPools);
-        for (Pool pool : poolsToDelete) {
-            this.sink.queueEvent(this.eventFactory.poolDeleted(pool));
-        }
+        firePoolDeletedEvents(poolsToDelete);
         log.info("Starting batch delete of entitlements");
         entitlementCurator.batchDelete(entsToRevoke);
```

Comment:
I'm confused. Your comment says "Replace batch delete of entitlements with batch delete by ids", but this is unchanged (it's still deleting by entity)?

Reply:
That is the issue I mentioned during scrum. With delete by ids I am hitting issues with detached entities. Not sure why that happens so I removed it from PR until it is fixed.